src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6705,17 +6705,47 @@ bool canBeNullableType()
                                 continue;
                         }
                         goto done; // token not consumed
+                    case SyntaxKind.OpenBracketToken when (nameOptions & NameOptions.PossiblePattern) != 0:
+                        // We don't want to consume a sized rank specifier in patterns, we leave it out to parse it as a length pattern.
+                        {
+                            if (isOmittedSize())
+                            {
+                                bool sawOpenBracket;
+                                var ranks = _pool.Allocate<ArrayRankSpecifierSyntax>();
+                                try
+                                {
+                                    do
+                                    {
+                                        ranks.Add(this.ParseArrayRankSpecifier(out _));
+                                    }
+                                    while ((sawOpenBracket = this.CurrentToken.Kind == SyntaxKind.OpenBracketToken) && isOmittedSize());
+
+                                    type = _syntaxFactory.ArrayType(type, ranks);
+                                }
+                                finally
+                                {
+                                    _pool.Free(ranks);
+                                }
+
+                                // If we saw an open bracket that is not followed by an omitted size, it's possibly a length pattern.
+                                if (!sawOpenBracket)
+                                    continue;
+                            }
+                            goto done;
+
+                            bool isOmittedSize() => this.PeekToken(1).Kind is SyntaxKind.CommaToken or SyntaxKind.CloseBracketToken;
```

Comment:
Please move local function at end of method

Reply:
We already have a local function in the middle of this method. Isn't that helpful for code locality, as long as it doesn't interfere with the actual code flow?

Reply:
Yeah, sometimes they slip through and it's not too intrusive to the code flow in your case. However, if possible (no scoped captures) we do prefer them at the end by convention (to know where to look to find them)